### PR TITLE
MemMap Fixes

### LINKED ISSUE #430: [opencl] hipStreamPerThread_basic failing for OpenCL dGPU
The following change in the test makes the test pass:

```
//   HIP_CHECK(hipHostMalloc(&hostMem, size));
  HIP_CHECK(hipMalloc(&devMem, size));
```

### PR DESCRIPTION
**Pull request by pvelesko**
* ~~Fix assert that event use after delete (recycled events would not get their `Deleted_` reset~~
* Registered memory map/unmap was not getting applied to async memcpy
* Registered memcopy mapped to `HOST_READ` in all cases instead of `HOST_WRITE` seems that `HOST_WRITE` doesn't guarantee a proper read whereas using `HOST_READ` allows to both write and read host pointers. 

fixes #430

### CODE REVIEW

File: src/backend/Level0/CHIPBackendLevel0.hh [hunk 1]
```diff
@@ -305,8 +305,10 @@ public:
     // go through all pools and try to get an allocated event
     for (size_t i = 0; i < EventPools_.size(); i++) {
       CHIPEventLevel0 *Event = EventPools_[i]->getEvent();
-      if (Event)
+      if (Event) {
+        Event->Deleted_ = false;
```
Comment (linehill) [inline src/backend/Level0/CHIPBackendLevel0.hh:309]: This is redundant. The `Deleted_` flag is reset by the `getEvent()`call in the line 307 (LZEventPool::getEvent() -> CHIPEventLevel0::reset() -> CHIPEventLevel0::markDeleted(false)).

Comment (pvelesko) [reply src/backend/Level0/CHIPBackendLevel0.hh:309]: wasn't working for Release, I believe. I was getting "event use after delete" errors. I need to clean up this branch, though.

File: src/CHIPBackend.hh [hunk 2]
```diff
@@ -607,10 +607,7 @@ protected:
   CHIPEventFlags Flags_;
   std::vector<CHIPEvent *> DependsOnList;
 
-#ifndef NDEBUG
-  // A debug flag for cathing use-after-delete.
-  bool Deleted_ = false;
-#endif
+
```
Comment (linehill) [inline src/CHIPBackend.hh:610]: This is not needed if (1) is reverted.

File: src/CHIPBackend.hh [hunk 3]
```diff
@@ -629,6 +626,10 @@ protected:
   CHIPEvent() = default;
 
 public:
+// #ifndef NDEBUG
+  // A debug flag for cathing use-after-delete.
+  bool Deleted_ = false;
+// #endif
```
Comment (linehill) [inline src/CHIPBackend.hh:632]: This is not needed if (1) is reverted.

**Pull request merged**